### PR TITLE
Disable tests which make network calls

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
By definition unit tests should not be making network calls. The current suite of tests consistently make us hit github rate limits. The network calls should be mocked out, but at the very least we should disable these from unit test runs.

## Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Plugin
- [x] Housekeeping

## Are all requirements met?

- [ ] Code completed
- [x] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2384

## Follow-up issue
_NA_

### CODE REVIEW

**Review by EngHabu — APPROVED** [review]

**Review by eapolinario — COMMENTED** [review]
Just for my own understanding, how did you figure out which tests were making network calls?

**Review by eapolinario — COMMENTED** [review]
Can you also add a new step to the gh workflow to invoke this new make target in CI?

**Comment by katrogan** [comment]
hey @eapolinario i debated about this while drafting the PR but if we run the tests on every merge to master even that's still making a fair amount of GH api calls. I'm not sure how to have a less consistent cadence for running these tests without hammering github

**Comment by katrogan** [comment]
closing in favor of https://github.com/flyteorg/flytectl/pull/315
